src/react/fileUrlToFileUrlObject.ts

```diff
@@ -5,32 +5,63 @@ import { getNtrData } from '../shared/ntrData'
 import {
   anyDynamicFilepathPartRegex,
   dynamicFilepathPartsRegex,
+  getCatchAllPathPartKey,
   getDynamicPathPartKey,
+  getOptionalCatchAllPathPartKey,
+  matchAllFilepathPartsRegex,
   optionalMatchAllFilepathPartRegex,
   spreadFilepathPartRegex,
 } from '../shared/regex'
 import type { TRouteBranch } from '../types'
 import { parseUrl } from './parseUrl'
 
+/**
+ * Get the route branch name from a route branch based on the locale or the name if no value for locale is found
+ */
+const getRouteBranchName = (routeBranch: TRouteBranch, locale: string) => {
+  if (!routeBranch?.name && routeBranch?.paths?.[locale]) {
+    return ''
+  }
+
+  if (routeBranch?.paths?.[locale]) {
+    return routeBranch?.paths?.[locale]
+  }
+
+  if (routeBranch?.name) {
+    return routeBranch?.name
+  }
+
+  return ''
+}
+
 /**
  * Recursively get path file UrlObject from a route branch
  */
 const getFileUrlObject = ({
   routeBranch,
   pathParts,
+  locale,
 }: {
   routeBranch: TRouteBranch
   /** Remaining path parts after the `routeBranch` path parts */
   pathParts: string[]
+  locale: string
 }): { pathname: string; query?: ParsedUrlQuery } => {
+  const routeBranchName = getRouteBranchName(routeBranch, locale)
```

Comment:
I know that was your initial approach but we need to translate the route and this is the best place to do so. If we don't translate the route the you end up with the problem that this pr is trying to fix.
Currently in this library if you have a route like '/men/shoes' and you are using 'es-ES' for example as locale the I would expect the link to take me to `/hombres/shoes` but that's not the case since the current behaviour is that when you click on the link it will take you to '/men/shoes' and after refreshing it will correctly re-direct me to `/hombres/shoes` thus you need to have this locale translation logic inside here.

You stated that you don't understand why the suggested changes solve my issue, well, is because the library is not translating the paths as it should. If you go to the [code sanbox demo](https://codesandbox.io/p/sandbox/next-translate-routes-dx3r8l?file=%2Fcomponents%2FLayout.tsx%3A1%2C1-36%2C1) that I created for this PR and swap between this library current link component and my updated one you will understand better what I'm talking about 😊